File: Arrays/Two Dimensional Arrays/magicalPark.py
m = int(input())
n = int(input())
k = int(input())
s = int(input())

park = [[(input()) for x in range(m)] for y in range(n)]

def magicalPark(arr, m, n, k, s):

    success = True

    for i in range(0, m):
        for j in range(0, n):
            ch = arr[i][j]

            if (s < k):
                break

            if (ch == '*'):
                s += 5
            elif (ch == '.'):
                s -= 2
            else:
                break

            if (j != n - 1):
                s -= 1
    if (success):
        print("yes", s)
    else:
        print("No")

magicalPark(park, m, n, k, s)

File: Arrays/Two Dimensional Arrays/magicalPark.cpp
#include <bits/stdc++.h>
using namespace std;

void magicalPark(char arr[][100], int m, int n, int k, int s){
    bool success = true;

    for (int i = 0; i < m; i++){
        for (int j = 0; j < n; j++){
            
            char ch = arr[i][j];
            //base case
            if (s < k){
                break;
            }

            //case 1
            if (ch == '*'){
                s += 5;
            }else if(ch == '.'){
                s -= 2;
            }else{
                break;
            }
            if (j != n - 1){
                s -= 1;
            }
        }
    }
    if (success){
        cout << "Yes, won with strength remaining - " << s << endl;
    }else{
        cout << "Lost" << endl;
    }
}

int main(){
    int m, n, k, s;
    cin >> m >> n >> k >> s;

    char park[100][100];

    for (int i = 0; i < m; i++){
        for (int j = 0; j < n; j++){
            cin >> park[i][j];
        }
    }

    magicalPark(park, m, n, k, s);



    return 0;
}
